guardrails/utils/pydantic_utils.py

```diff
@@ -0,0 +1,172 @@
+"""
+RAIL File:
+
+Version A:
+    <pydantic 
+        name="person" 
+        description="Information about a person." 
+        model="Person" 
+        on-fail-{root-validator-name}="blah" 
+        # on-fail-{field-name}-{validator-name}="blah" 
+        on-fail-age-between-0-150="reask" 
+    />
+where Person is a Pydantic BaseModel.
+
+    <pydantic 
+        model="Person" 
+        name="person" 
+        description="Information about a person." 
+        on-fail-{root-validator-name}="blah" 
+    >
+        # Omit fields that don't need a custom description or on-fail
+        <field name="age" description="The age of the person." on-fail-between-0-150="reask" />
+    </pydantic>
+
+Which is converted to:
+
+    <object name="person" description="Information about a person." model="Person">
+        <string name="name" description="The name of the person." on-fail-{validator-name}="blah" />
+        <integer name="age" description="The age of the person." on-fail-between-0-150="reask" />
+    </object>
+
+internally.
+"""
+from typing import TYPE_CHECKING, Dict
+
+import lxml
+from griffe.dataclasses import Docstring
+from griffe.docstrings.parsers import Parser, parse
+
+if TYPE_CHECKING:
+    from pydantic import BaseModel
+
+
+def get_field_descriptions(model: BaseModel) -> Dict[str, str]:
+    """Get the descriptions of the fields in a Pydantic model using the docstring."""
+    docstring = Docstring(model.__doc__, lineno=1)
+    parsed = parse(docstring, Parser.google)
+
+    # TODO: change parsed[1] to an isinstance check for the args section
+    return {
+        field.name: field.description.replace("\n", " ")
+        for field in parsed[1].as_dict()["value"]
+    }
+
+
+PYDANTIC_SCHEMA_TYPE_MAP = {
```

Comment:
pydantic supports many more types though https://docs.pydantic.dev/usage/types/

Reply:
Currently, the code does allow the developer to use an arbitrary BaseModel (any Pydantic types they like). But there's a question as you point out about how to handle their model correctly. 

This particular `PYDANTIC_SCHEMA_TYPE_MAP` mapping is about converting the Pydantic type into a `RAIL` element (e.g. a `file: FilePath` field -----> `<string ... />`). 

All Pydantic types are mapped to JSON versions (https://docs.pydantic.dev/usage/schema/#json-schema-types) when one calls `MyModel(...).schema()`: this `PYDANTIC_SCHEMA_TYPE_MAP` dictionary converts those JSON types into Guardrail's currently implemented elements. (It may be worth aligning these so this dictionary is not required in another PR.)

To answer your question now: I was planning on adding `format="file-path"` to the converted RAIL element, so `<string format="file-path;" />`. This will make the LLM is aware of the Pydantic type in the prompt, and this information is also easy to get from either the `MyModel(...).schema()` (or just the field definitions themselves).